src/mongo/tree/MongoCollectionTreeItem.ts

```diff
@@ -107,41 +107,36 @@ export class MongoCollectionTreeItem extends AzureParentTreeItem<IMongoTreeRoot>
 		return new MongoDocumentTreeItem(this, newDocument);
 	}
 
-	executeCommand(name: string, args?: string[]): Thenable<string> | null {
+	public async tryExecuteCommandDirectly(name: string, args?: string[]): Promise<{ deferToShell?: boolean; result?: string }> {
```

Comment:
The original code was really weird and had to understand/debug - it returned a promise if it ran directly, or null (not a promise returning null) if you should run it in the shell instead.